modules/ethereum/src/verification.rs

```diff
@@ -176,13 +321,55 @@ fn verify_signature(expected_validator: &Address, signature: &H520, message: &H2
 		.unwrap_or(false)
 }
 
+/// Find next unfinalized validators set change after finalized set.
+fn find_next_validators_signal<S: Storage>(
+	storage: &S,
+	context: &ImportContext<S::Submitter>,
+	header: &Header,
+) -> Option<Vec<Address>> {
+	// that's the earliest block number we may met in following loop
+	// it may be None if that's the first set
+	let best_set_signal_block = context.validators_set().signal_block;
+
+	// if parent schedules validators set change, then it may be our set
+	// else we'll start with last known change
+	let parent_schedules_change = context.parent_scheduled_change();
+	let mut current_set_signal_block = match parent_schedules_change {
```

Comment:
Second time I see that logic, maybe it's worth to make it a function on `context`?